bears/vcs/git/GitCommitBear.py

```diff
@@ -203,3 +243,82 @@ def check_body(self, body,
+                              body_close_issue_full_url: bool=False,
+                              body_close_issue_on_last_line: bool=False):
+        """
+        Check for matching issue related references and URLs.
+
+        :param body:
+            Body of the commit message of HEAD.
+        :param body_close_issue:
+            Whether to check for the presence of issue reference within
+            the commit body by retrieving host information from git
+            configuration. GitHub and GitLab support auto closing issues with
+            commit messages. Checks for matching keywords in the commit body.
+            By default, if none of ``body_close_issue_full_url`` and
+            ``body_close_issue_on_last_line`` are enabled, checks for presence
+            of short references like ``closes #213``. Otherwise behaves
+            according to other chosen flags.
+            More on keywords follows.
+            [GitHub](https://help.github.com/articles/closing-issues-via-commit-messages/)
+            [GitLab](https://docs.gitlab.com/ce/user/project/issues/automatic_issue_closing.html)
+        :param body_close_issue_full_url:
+            Checks the presence of issue close reference with a full URL
+            related to some issue. Works along with ``body_close_issue``.
+        :param body_close_issue_on_last_line:
+            When enabled, checks for issue close reference presence on the
+            last line of the commit body. Works along with
+            ``body_close_issue``.
+        """
+        if not body_close_issue:
+            return
+
+        host = self.get_host_from_remotes()
+        if host not in self.SUPPORTED_HOST_KEYWORD_REGEX:
+            return
+
+        if body_close_issue_on_last_line:
+            body = body.splitlines()[-1]
+            result_message = ('Body of HEAD commit does not contain any {} '
+                              'reference in the last line.')
+        else:
+            result_message = ('Body of HEAD commit does not contain any {} '
+                              'reference.')
+
+        if body_close_issue_full_url:
+            result_info = 'full issue'
+            compiled_issue_ref_regex = re.compile(
+                r'https?://{}\S+/issues/\w+'.format(re.escape(host)))
+        else:
+            result_info = 'issue'
+            compiled_issue_ref_regex = re.compile(r'#\w+')
+
+        concat_regex = '|'.join(kw for kw in self.CONCATENATION_KEYWORDS)
+        compiled_joint_regex = re.compile(
+            r'(?:{0})\s+'
+            r'((?:\S(?!{1}))*\S(?:\s*(?:{1})\s*(?!{0})(?:\S(?!{1}))*\S)*)'
+            r''.format(
+                self.SUPPORTED_HOST_KEYWORD_REGEX[host],
+                concat_regex))
+
+        matches = compiled_joint_regex.findall(body)
+
+        if len(matches) == 0:
+            yield Result(self, result_message.format(result_info))
+            return
+
+        compiled_issue_num_regex = re.compile(r'(?:#|\D+/)[1-9][0-9]*')
```

Comment:
I think you could use the regex without the `(?:#|\D+/)` part. You could add a capturing group inside `compiled_issue_ref_regex` getting the right part of the reference that represent the issue number. Inside the check-loop below, you extract the group and just try a fullmatch with `[1-9][0-9]*`.

what do you think?